### DISCUSSION TITLE
RocketMQ v5 是否有计划废除 Remoting 协议？

### DISCUSSION THREAD

**Discussion by zergduan** [apache/rocketmq#8745] [Q&A] [answered]
RocketMQ v5 开始推出了 gRPC 协议的 SDK，我想请教几个相关问题：

1. 使用 gRPC 协议的 Client 只能连接 Proxy 架构的 RocketMQ 集群，对吧？
2. 使用 Remoting 协议的 Client 不需要连接 Proxy ，依然使用 v4 版本的方式直接连接 NameServer和Broker对吧？
3. 如果只有 Remoting 协议 Client需要连接集群，则集群不需要启动 Proxy 组件，对吧？也就是说在RocketMQv5结群使用传统架构（v4架构：NameServer+Broker主备）依然可以给Remoting Client提供服务？
4. 目前社区是否由计划要废除 Remoting SDK 的支持？如果有着方面的考虑，大概终止支持的时间点是什么时候？

以上的问题在于当前我们的使用环境仅限于java客户端，且Remoting SDK使用成熟且满足所有需求，并不打算做客户端变更；但是也要考虑到未来的Remoting SDK支持周期，所以希望提前得到信息，来计划未来是否做相关切换。

谢谢

**Comment by leizhiyuan** [comment, marked as answer]
1. 是的
2.  可以
3. 可以
4. 没有

**Reply by zergduan** [reply]
> 1. 是的
> 2. 可以
> 3. 可以
> 4. 没有

感谢您的解答
